1 - DataTypes/Code/WeirdCalculator.cs

```diff
@@ -39,23 +42,34 @@ public static decimal ComplexDecimalAdd(float firstValue, double secondValue, de
 			 * I will give you one clue / tip - You *will* need to handle exceptions for the final solution.
 			 * If you are new to exceptions, google "C# Exception Handling" - litterally half our jobs is googling :P
 			 */
-
-			throw new NotImplementedException();
+            decimal result=0;
+            try
+            {
+               result = (decimal)firstValue + (decimal)secondValue + thirdValue;
+            }
+            catch (OverflowException)
+            {
+                 result = 0;
+            }
+            return result;
+            
+            throw new NotImplementedException();
 		}
 
 		public static decimal ComplexAdd(char firstValue, string secondValue)
 		{
-			/*
+            /*
 			 * You will notice the TDD tests around this just "assume" it will work. Do not worry about handling exceptions or errors, assume input is valid.
 			 */
-
+            var result = decimal.Parse(char.ToString(firstValue))+decimal.Parse(secondValue); //this seems messy but since ToDecimal(Char) is not supported I couldn't think of any other way
```

Comment:
Though this does work, its fairly slow to go from char -> string -> decimal. The convert method is what you are looking for:

[Convert](https://docs.microsoft.com/en-us/dotnet/api/system.convert.todecimal?view=netframework-4.8#System_Convert_ToDecimal_System_Char_)